pkg/asset/cluster/tfvars.go

```diff
@@ -300,6 +300,10 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 				securityGroups = append(securityGroups, mp.Platform.AWS.AdditionalSecurityGroupIDs...)
 			}
 		}
+		// Use security groups from defaultMachinePlatform if not specified in the controlPlane stanza
+		if mp := installConfig.Config.AWS.DefaultMachinePlatform; mp != nil && len(securityGroups) == 0 {
+			securityGroups = append(securityGroups, mp.AdditionalSecurityGroupIDs...)
```

Comment:
Do we have to check if [AdditionalSecurityGroupIDs ](https://github.com/openshift/installer/blob/master/data/data/install.openshift.io_installconfigs.yaml#L2256) is specified before appending them to `securityGroups`?

Reply:
No. `append` works with nil slices. E.g.: https://go.dev/play/p/clRsPPWkQkD?v=goprev

Reply:
Shouldn't it be like use from default first and overwrite from control plane stanza? Only asking since we follow this everywhere else

Reply:
Hmm, that makes sense. Let me look into it.

Reply:
@rna-afk should be better now.